extensions/guacamole-vault/modules/guacamole-vault-ksm/src/main/java/org/apache/guacamole/vault/ksm/secret/KsmRecordService.java

```diff
@@ -229,6 +231,10 @@ private <T> String getSingleStringValue(List<T> values, Function<T, String> mapp
     private <T extends KeeperRecordField> T getField(List<KeeperRecordField> fields,
             Class<T> fieldClass, Pattern labelPattern) {
 
+        // There are no fields if no List was provided at all
+        if (fields == null)
```

Comment:
I'm not suggesting that you do this right now, since it's a bit of a refactor-cat-in-bathtub thing, but having `@Nullable` and `@Nonnull` annotations is pretty handy for this sort of thing. The compiler would complain if you try to pass anything that might be null in here.

Reply:
I'm actually surprised the compiler didn't catch this, as `getCustom()` is annotated with `@Nullable`.

Reply:
It would be nice if it worked like that, but as far as I can tell, it needs the explicit `@NotNull` for the compiler to do anything about it.